### DISCUSSION TITLE
map method for showing multiple bookings not working

### DISCUSSION THREAD

**Discussion by Charlotte-Ketteler** [neuefische/questions#683] [Web]
Hi everyone,
in my booking app, I've managed to book an appointment with date, numberOfPeople and the horses for a riding lesson. The booking gets displayed on my BookingOverviewPage, where I can also delete the booking. Now I want to display multiple bookings. I looked at the example: [React State 3](https://codesandbox.io/s/react-state-3-demo-61efn5?file=/src/App.js) which I think is a good example for my intent. Now I already applied code in my app.js in the handleSubmit function (line 22 and 47), and used the map function in my BookingOverviewPage. Unfortunately I get an error saying: TypeError: Cannot read properties of null (reading 'map'). Meaning recentlyBooked. Before I implemented map, it worked, though using the delete button twice I got the same error which I don't understand. Do you think I'm on the right path with my code and do you have any suggestions? [Branch: Terminübersicht](https://github.com/Charlotte-Ketteler/capstone-project/tree/Termin%C3%BCbersicht)

Maybe you can help, cheers! :)

**Comment by ninaassmann** [comment]
line 47
setRecentlyBooked([ ...booking, {...newBooking, bookingID }]);

where do you get the booking and newBooking?

and with that you overwrite the whole recentlyBooked, so i guess it will always be just one entry

**Reply by Charlotte-Ketteler** [reply]
Hi Nina, thanks for looking into my discussion. I changed my handleSubmit in app.js now and left out the newBookings and instead I am using timeSlots. It's still not working as I can't solve the issue with recentlyBooked is undefined. I guess I have to try out some other ways. 

![Screenshot 2023-08-20 at 15 33 51](https://github.com/neuefische/questions/assets/127936697/3aed2d61-72b4-4ccd-98dd-613c1f586cd7)
![Screenshot 2023-08-20 at 15 34 13](https://github.com/neuefische/questions/assets/127936697/d455fabe-96fb-462c-9010-f70815bf471a)
